hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieFileGroup.java

```diff
@@ -62,14 +59,16 @@
   /**
    * The last completed instant, that acts as a high watermark for all getters
    */
-  private final Optional<HoodieInstant> lastInstant;
+  private final com.google.common.base.Optional<HoodieInstant> lastInstant;
 
   public HoodieFileGroup(String partitionPath, String id, HoodieTimeline timeline) {
     this.partitionPath = partitionPath;
     this.id = id;
     this.fileSlices = new TreeMap<>(HoodieFileGroup.getReverseCommitTimeComparator());
     this.timeline = timeline;
-    this.lastInstant = timeline.lastInstant();
+    // Guava Optional is serializable.
+    // TODO : varadarb -> Open another PR to replace java.util.Optional with Guava'a everywhere internally
```

Comment:
Open to trying it.. but Guava is tricky.. it usually has ton of version conflicts.. Roll our own instead?

Reply:
I have only one concern with using our own Optional type -> If any of these classes are exposed to external users, then for them to understand a custom implementation of Optional is an overhead. Most people are used to standard Guava or Java optionals...

Reply:
Went ahead and made this change.